src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -618,15 +626,14 @@ void updateInCache
                                           mdStringFilterP,
                                           doc.hasField(CSUB_FORMAT)? stringToRenderFormat(getStringFieldF(doc, CSUB_FORMAT)) : NGSI_V2_NORMALIZED);
 
-  if (subCacheP != NULL)
-  {
-    LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
-    subCacheItemRemove(subCacheP);
-  }
-
   if (mscInsert == 0)  // 0: Insertion was really made
   {
     subCacheUpdateStatisticsIncrement();
+    if (subCacheP != NULL)
+    {
+      LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
+      subCacheItemRemove(subCacheP);
```

Comment:
Moving the remove operation inside the `if (mscInsert == 0)` is not directly related with the fix implemented in this bug. However, I think is an important "safety measure" as in the future other conditions may fail at mongoSubCacheItemInsert() and we don't want in that case not only not updating the subscription but removing it from the cache.